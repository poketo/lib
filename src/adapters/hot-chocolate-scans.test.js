import poketo from '..';
import adapter from './hot-chocolate-scans';

describe('HotChocolateScansAdapter', () => {
  describe('supportsUrl', () => {
    it('returns true for urls like hotchocolatescans.com', () => {
      expect(adapter.supportsUrl('http://hotchocolatescans.com')).toBe(true);
      expect(adapter.supportsUrl('http://hotchocolatescans.com/fs')).toBe(true);
      expect(adapter.supportsUrl('http://hotchocolatescans.com/fs/a/b')).toBe(
        true,
      );
    });

    it('returns false for urls that are not hotchocolatescans.com', () => {
      expect(adapter.supportsUrl('http://he.lveticascans.com')).toBe(false);
    });
  });

  describe('parseUrl', () => {
    it('returns the components of a url', () => {
      expect(
        adapter.parseUrl(
          'http://hotchocolatescans.com/fs/series/itoshi-no-muco/',
        ),
      ).toEqual({
        seriesSlug: 'itoshi-no-muco',
        chapterSlug: null,
      });

      expect(
        adapter.parseUrl(
          'http://hotchocolatescans.com/fs/read/itoshi-no-muco/en/1/2/page/1',
        ),
      ).toEqual({ seriesSlug: 'itoshi-no-muco', chapterSlug: 'en/1/2' });

      expect(
        adapter.parseUrl(
          'http://hotchocolatescans.com/fs/read/mousou-telepathy/en/0/512/5/page/25',
        ),
      ).toEqual({ seriesSlug: 'mousou-telepathy', chapterSlug: 'en/0/512/5' });
    });

    it('throws on unparseable paths', () => {
      expect(() => {
        adapter.parseUrl(
          'http://hotchocolatescans.com/fs/other/mousou-telepathy/en/2/11/page/1',
        );
      }).toThrow(poketo.InvalidUrlError);

      expect(() => {
        adapter.parseUrl('http://hotchocolatescans.com/fs/');
      }).toThrow(poketo.InvalidUrlError);
    });
  });
});

describe('HotChocolateScans', () => {
  const server = new AdapterVcrServer(adapter);

  beforeAll(async () => {
    await server.listenAndMock();
  });

  afterAll(() => {
    server.close();
  });

  describe('getSeries', () => {
    it('returns a metadata object', async () => {
      const { chapters, ...metadata } = await poketo.getSeries(
        'hot-chocolate-scans:watashi_no_shounen',
      );

      expect(metadata).toMatchSnapshot({
        updatedAt: expect.any(Number),
      });

      const chapterNumbersToTest = ['4', '8'];
      const chaptersToTest = chapters.filter(chapter =>
        chapterNumbersToTest.includes(chapter.chapterNumber),
      );
      expect(chaptersToTest).toMatchSnapshot();
    });
  });

  describe('getChapter', () => {
    it('returns a list of pages', async () => {
      const chapter = await poketo.getChapter(
        'hot-chocolate-scans:watashi_no_shounen:en/0/4',
      );

      expect(chapter.pages).toHaveLength(33);
      expect(chapter.pages).toEqual(
        expect.arrayContaining([
          {
            id: expect.any(Number),
            width: expect.any(Number),
            height: expect.any(Number),
            url: expect.stringContaining('http'),
          },
        ]),
      );
    });
  });
});
