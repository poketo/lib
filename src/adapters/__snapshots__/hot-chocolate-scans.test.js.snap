// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HotChocolateScans getSeries returns a metadata object 1`] = `
Object {
  "author": "Takano Hitomi",
  "coverImageUrl": "http://hotchocolatescans.com/fs/content/comics/watashi_no_shounen_5a68c187815a3/thumb_001.jpg",
  "description": "Satoko Tawada is an office worker. She's almost 30 years old, but she still hasn't been able to find happiness in her life. Fate brings her with Mashuu Hayami, a beautiful 12-year-old boy whom she finds practicing soccer all by himself in the park. Out of a whim, she decides to help Mashuu with his practice, with her knowledge of soccer at her disposal. As they practice together on an everyday routine, the two eventually develop an emotional bond. When Satoko's ex-boyfriend deceives her, Mashuu gladly comforts her. And when Satoko learns of Mashuu's mostly absent parents, she starts to concern herself with his personal life. The two lonely souls find common ground and develop an emotional bond that transcends their age difference and circumstances. With this newfound bond, new feelings start to sprout within Satoko. Are these feelings that of maternity? Or is it something else?",
  "id": "hot-chocolate-scans:watashi_no_shounen",
  "publicationStatus": "UNKNOWN",
  "site": Object {
    "id": "hot-chocolate-scans",
    "name": "Hot Chocolate Scans",
  },
  "slug": "watashi_no_shounen",
  "supportsReading": true,
  "title": "Watashi no Shounen",
  "updatedAt": Any<Number>,
  "url": "http://localhost:57155/fs/series/watashi_no_shounen",
}
`;

exports[`HotChocolateScans getSeries returns a metadata object 2`] = `
Array [
  Object {
    "chapterNumber": "8",
    "createdAt": 1518828182,
    "id": "hot-chocolate-scans:watashi_no_shounen:en/0/8",
    "slug": "en/0/8",
    "url": "http://localhost:57155/fs/read/watashi_no_shounen/en/0/8/page/1",
    "volumeNumber": "0",
  },
  Object {
    "chapterNumber": "4",
    "createdAt": 1516818687,
    "id": "hot-chocolate-scans:watashi_no_shounen:en/0/4",
    "slug": "en/0/4",
    "url": "http://localhost:57155/fs/read/watashi_no_shounen/en/0/4/page/1",
    "volumeNumber": "0",
  },
]
`;
